Automattic/newspack-popups
# feat: `seen` events for GA4

### All Submissions:






### Changes proposed in this Pull Request:



Implements `seen` events for GA4 when prompts become visible in the viewport. This PR simply implements the logic for firing the event and logs the event payload to the dev console when fired; actually sending the events to GA will come in a future PR.

Closes `1204616440912018/1204751246643878`.

### How to test the changes in this Pull Request:

1. Check out this branch.
2. Open your dev console and view a page which will display one or more prompts.
3. Confirm that the `seen` event for each prompt gets logged to the dev console once at least 50% of the prompt is visible on screen for at least 250 milliseconds.
4. Confirm that each event fires only once per prompt per pageview, even if you scroll past the prompt and then scroll back to make it visible again.

### Other information:
